[core] Add call_site to Tasks and Actors. (#48920)
ray-project/ray: doc/source/ray-core/tasks.rst

Nested tasks don't inherit the task events settings from the parent task. You need to set the task events settings for each task separately.

> Invocation Stacktrace

Reviewer: ```suggestion
Stacktrace invocation
```

---------------------

> Ray can optionally capture and display the stacktrace of where tasks are invoked from in your code. This can help with debugging and understanding the execution flow of your application.

Reviewer: ```suggestion
Ray can optionally capture and display the stacktrace of where your code invokes tasks. This feature can help with debugging and understanding the execution flow of your application.
```

To enable invocation stacktrace capture, set the environment variable ``RAY_enable_invocation_stacktrace=true``. When enabled:

> - The stacktrace will be captured when creating and invoking tasks

Reviewer: ```suggestion
- Ray captures the stacktrace when creating and invoking tasks
```

> - The stacktrace will be visible in:

Reviewer: ```suggestion
- The stacktrace is visible in:
```

  - Ray Dashboard UI under the task details
  - ``ray list task --detail`` CLI command output
  - State API responses

Note that stacktrace capture is disabled by default to avoid any performance overhead. Only enable it when needed for debugging purposes.

Example:

.. testcode::

    import ray

    # Enable stacktrace capture
    ray.init(runtime_env={"env_vars": {"RAY_enable_invocation_stacktrace": "true"}})

    @ray.remote
    def my_task():
        return 42

>     # Stacktrace will be captured for task invocation

Reviewer: ```suggestion
    # Capture the stacktrace upon task invocation
```

    future = my_task.remote()
    result = ray.get(future)

> The stacktrace will show the exact line numbers and call stack where the task was created and invoked.

Reviewer: ```suggestion
The stacktrace shows the exact line numbers and call stack where the task was created and invoked.
```

> This feature is currently only supported for Python and C++ tasks.

Reviewer 2: Instead of repeating the basic same thing for actor and task, I think the better place to have this is under https://docs.ray.io/en/latest/ray-observability/user-guides/debug-apps/index.html

Author: good point, moved here
